com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/flowable/FlowableFacade.java

```diff
@@ -268,6 +280,25 @@ public boolean isProcessInstanceAtReceiveTask(String processInstanceId) {
                                    .collect(Collectors.toList());
     }
 
+    private List<Execution> getAllLeafExecutions(String processInstanceId) {
+        List<Execution> allExecutions = getAllProcessExecutions(processInstanceId);
+        return allExecutions.stream()
```

Comment:
This is a lot complexity: O(n^2).
Could we revert the things like this:
1. Find the execution which is related to the processInstanceId
2. Find each executions which have parentId = processInstaceId || superExecutionId = processInstanceId
3. Do 1 and 2, recursively for each execution foud in the step 2 
Would this work?
This way we would lower the complexity to O(nlogn)

Reply:
About the complexity I do agree.
However, I am not sure if it worths to have recursion here in order to optimize a little bit when the code will not look nicer.